machine/accton/accton_as_28t_ops/kernel/platform-as_28t_ops.patch

```diff
@@ -0,0 +1,479 @@
++                reg = <0x00000000 0x000d0000>;
++                label = "uboot";
++            };
++            partition@1 {
++                /* shmoo - 64KB*/
++                reg = <0x000d0000 0x00010000>;
++                label = "shmoo";
++            };
++            partition@2 {
++                /* uboot-env - 64KB */
++                reg = <0x000e0000 0x00010000>;
++                label = "uboot-env";
++                env_size = <0x02000>;
++            };
++            partition@3 {
++                /* onie - 7MB */
++                reg = <0x000f0000 0x00700000>;
++                label = "onie";
++            };
++            partition@4 {
++                /* hw-info - 64KB */
++                reg = <0x007f0000 0x00010000>;
++                label = "hw-info";
++            };
++            partition@5 {
++                /* diag - 24MB */
++                reg = <0x00800000 0x01800000>;
++                label = "diag";
++            };
++        };
++    };
++};
+diff --git a/arch/arm/mach-iproc/board_bu.c b/arch/arm/mach-iproc/board_bu.c
+index b45588d..6142fdb 100644
+--- a/arch/arm/mach-iproc/board_bu.c
++++ b/arch/arm/mach-iproc/board_bu.c
+@@ -1056,6 +1056,25 @@ MACHINE_START(IPROC, "Broadcom iProc")
+ 	.init_machine = board_init,
+ MACHINE_END
+ 
++static const char * hurricane2_dt_board_compat[] = {
++    "accton,as_28t_ops",
++    NULL
++};
++
++DT_MACHINE_START(HURRICANE2_DT, "Broadcom Hurricane2 (Flattened Device Tree)")
++	.map_io     = board_map_io,
++	.init_early = iproc_init_early,
++	.init_irq   = iproc_init_irq,
++	.handle_irq     = gic_handle_irq,
++	.timer      = &board_timer,
++	.init_machine   = board_init,
++	.dt_compat  = hurricane2_dt_board_compat,
++MACHINE_END
++
++static const char * helix4_dt_board_compat[] = {
++    NULL
++};
++
+ DT_MACHINE_START(HELIX4_DT, "Broadcom Helix4 (Flattened Device Tree)")
+ 	.map_io     = board_map_io,
+ 	.init_early = iproc_init_early,
+diff --git a/arch/arm/mach-iproc/flash.c b/arch/arm/mach-iproc/flash.c
+index 5eaf3ec..dec9d70 100644
+--- a/arch/arm/mach-iproc/flash.c
++++ b/arch/arm/mach-iproc/flash.c
+@@ -44,22 +44,32 @@
+ #if defined(CONFIG_IPROC_QSPI) || defined(CONFIG_IPROC_QSPI_MODULE)
+ static struct mtd_partition sflash_partition_map[] = {
+     {
+-        .name = "boot",
++        .name = "uboot",
+         .offset = 0x00000000,
+-        .size = 640 * 1024,
++        .size = 832 * 1024,
+     },
+     {
+-        .name = "env",
++        .name = "shmoo",
+         .offset = MTDPART_OFS_APPEND,
+-        .size = 384 * 1024,
++        .size = 64 * 1024,
+     },
+     {
+-        .name = "system",
++        .name = "uboot-env",
+         .offset = MTDPART_OFS_APPEND,
+-        .size = 15 * 1024 * 1024,
++        .size = 64 * 1024,
+     },
+     {
+-        .name = "rootfs",
++        .name = "onie",
++        .offset = MTDPART_OFS_APPEND,
++        .size = 7 * 1024 * 1024,
++    },
++    {
++        .name = "hw-info",
```

Comment:
Is this partition related to board_eeprom. If so, can we name it as board_eeprom for consistency across 4610 and 4510 platforms.